test/oli_web/live/delivery/instructor_dashboard/scored_activities/scored_activities_tab_test.exs

```diff
@@ -1241,6 +1241,64 @@ defmodule OliWeb.Delivery.InstructorDashboard.ScoredActivitiesTabTest do
       assert a4.title == "Module 2: BasicsPage 4"
     end
 
+    test "sorting by Due Date", %{
+      conn: conn,
+      section: section,
+      page_1: page_1,
+      page_2: page_2,
+      page_3: page_3
+    } do
+      base_time = ~U[2000-01-20 12:00:00.000000Z]
+      end_date_1 = base_time |> DateTime.add(1, :day) |> DateTime.truncate(:second)
+      end_date_2 = base_time |> DateTime.add(2, :day) |> DateTime.truncate(:second)
+      end_date_3 = base_time |> DateTime.add(3, :day) |> DateTime.truncate(:second)
+
+      Sections.get_section_resource(section.id, page_1.resource_id)
+      |> Sections.update_section_resource(%{end_date: end_date_1, scheduling_type: :due_by})
+
+      # Only section resources of scheduling_type = due_by are considered when sorting by Due Date
+      # The next SectionResource has scheduling_type: :read_by and it shouldn't be considered
+      Sections.get_section_resource(section.id, page_2.resource_id)
+      |> Sections.update_section_resource(%{end_date: end_date_2})
+
+      Sections.get_section_resource(section.id, page_3.resource_id)
+      |> Sections.update_section_resource(%{end_date: end_date_3, scheduling_type: :due_by})
```

Comment:
NIT: I think this might improve the readability of the test

```Elixir
      Sections.get_section_resource(section.id, page_1.resource_id)
      |> Sections.update_section_resource(%{
        end_date: ~U[2000-01-21 12:00:00.00Z],
        scheduling_type: :due_by
      })

      # Only section resources of scheduling_type = due_by are considered when sorting by Due Date
      # The next SectionResource has scheduling_type: :read_by and it shouldn't be considered
      Sections.get_section_resource(section.id, page_2.resource_id)
      |> Sections.update_section_resource(%{end_date: ~U[2000-01-22 12:00:00.00Z]})

      Sections.get_section_resource(section.id, page_3.resource_id)
      |> Sections.update_section_resource(%{
        end_date: ~U[2000-01-23 12:00:00.00Z],
        scheduling_type: :due_by
      })
```

Reply:
Interesting approach...

Another way to do it would be something like:

```
[
  {end_date: ~U[2000-01-21 12:00:00.00Z], scheduling_type: :due_by}
  {end_date: ~U[2000-01-22 12:00:00.00Z], scheduling_type: :read_by}
  {end_date: ~U[2000-01-23 12:00:00.00Z], scheduling_type: :due_by}
] 
|> Enum.each( fn {datetime, scheduling_type} ->
  Sections.get_section_resource(section.id, page_3.resource_id)
  |> Sections.update_section_resource(datetime, scheduling_type)
end
)
```

and maybe put the comment on top.